include/mbedtls/pk.h

```diff
@@ -170,6 +176,31 @@ typedef size_t (*mbedtls_pk_rsa_alt_key_len_func)( void *ctx );
 #endif /* MBEDTLS_PK_RSA_ALT_SUPPORT */
 
 /**
+ * \brief           Check whether instances of one PK-type are convertible
+ *                  to those of another.
+ *
+ * \param from      PK-type to convert from
+ * \param to        PK-type to convert to
+ *
+ * \note            Convertibility is currently restricted to the following two cases:
+ *                  (1) The target PK-type has the same underlying
+ *                      mathematical entity but is restricted in its
+ *                      functionality. For example, the type PK_ECKEY
+ *                      of unrestricted EC keys converts to PK_ECDSA
+ *                      of EC keys to use for ECDSA signing only, but
+ *                      not conversely.
+ *                  (2) The target PK-type has the same underlying
+ *                      mathematical entity but uses it differently
+ *                      for some of its functionality. For example,
+ *                      PK_RSA converts to PK_RSASSA_PSS which has uses
+ *                      different padding scheme for signatures.
+ *
+ * \return          0 if \c from is convertible to \c to, nonzero otherwise.
+ *
+ */
+int mbedtls_pk_type_convertible( mbedtls_pk_type_t from, mbedtls_pk_type_t to );
```

Comment:
No need to apologize, these things have to be consistent of course. I have no objections against either of the two changes you suggested. Which one do you think would be more suitable?